cpp/CMakeLists.txt

```diff
@@ -114,8 +114,8 @@ target_include_directories(basix PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR};${CMAKE_CURRENT_SOURCE_DIR}>")
 
 # TODO: remove after issue https://github.com/FEniCS/basix/issues/842 is resolved
-target_compile_definitions(basix PRIVATE MDSPAN_USE_PAREN_OPERATOR=1)
-target_compile_definitions(basix PRIVATE MDSPAN_USE_BRACKET_OPERATOR=0)
+target_compile_definitions(basix PUBLIC MDSPAN_USE_PAREN_OPERATOR=1)
+target_compile_definitions(basix PUBLIC MDSPAN_USE_BRACKET_OPERATOR=0)
```

Comment:
Not sure whether you want to deactivate the `operator[](...)` in the public interface. This might cause problems in user-code.